orchagent/vrrporch.cpp

```diff
@@ -0,0 +1,244 @@
+                                ip_pfx.to_string().c_str(),mac.to_string().c_str(),vrrp_rif_id);
+    }
+    else
+    {
+        SWSS_LOG_NOTICE("_vrrp_table entry already exists with vmac %s, rif_id %lx, update vmac to %s, rifid to %lx",
+				vrrp_table_[key].vmac.to_string().c_str(),vrrp_table_[key].rifid,
+				request.getAttrMacAddress("vmac").to_string().c_str(),vrrp_rif_id);
+        vrrp_table_.erase(key);
+        vrrp_table_[key] = {request.getAttrMacAddress("vmac"),vrrp_rif_id};
+    }
+	//Flush the FDB entry for vmac on vrrp master
+    FdbEntry entry;
+    entry.mac = request.getAttrMacAddress("vmac");
+    entry.bv_id = port_oid;
+    gFdbOrch->removeFdbEntry(entry, FDB_ORIGIN_LEARN);
+    return true;
+}
+bool VrrpOrch::delOperation(const Request& request)
+{
+    SWSS_LOG_ENTER();
+    Port port;
+    sai_object_id_t port_oid = 0;
+    auto key = vrrp_key_t(request.getKeyString(0),request.getKeyIpPrefix(1));
+    auto it = vrrp_table_.find(key);
+    if (it == vrrp_table_.end())
+    {
+        SWSS_LOG_ERROR("VRRP entry for port %s, vip %s doesn't exist", request.getKeyString(0).c_str(),request.getKeyIpPrefix(1).to_string().c_str());
+        return true;
+    }
+    gPortsOrch->getPort(request.getKeyString(0), port);
+    auto ip_pfx = request.getKeyIpPrefix(1);
+    sai_route_entry_t unicast_route_entry;
+    unicast_route_entry.switch_id = gSwitchId;
+    unicast_route_entry.vr_id = port.m_vr_id;
+    copy(unicast_route_entry.destination, ip_pfx.getIp());
+    switch(port.m_type)
+    {
+        case Port::PHY:
+            port_oid = port.m_port_id;
+            break;
+        case Port::LAG:
+            port_oid = port.m_lag_id;
+            break;
+        case Port::VLAN:
+            port_oid = port.m_vlan_info.vlan_oid;
+            break;
+        case Port::SUBPORT:
+            port_oid = port.m_parent_port_id;
+            break;			
+        default:
+            SWSS_LOG_ERROR("Unsupported port type: %d", port.m_type);
+            break;
+    }
+    //Flush the FDB entry for vmac
+    FdbEntry entry;
+    entry.mac = vrrp_table_[key].vmac;
+    entry.bv_id = port_oid;
+    gFdbOrch->removeFdbEntry(entry, FDB_ORIGIN_LEARN);
+	//If same vip is configured as DIP on the port, then skip delete vip as it will delete the DIP on the port.
+    if (!hasSameIpAddr (request.getKeyString(0), request.getKeyIpPrefix(1)))
+    {
+        sai_status_t vip_status = sai_route_api->remove_route_entry(&unicast_route_entry);
+        if (vip_status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to delete Vrrp Vip on interface %s, rv:%d",
+                port.m_alias.c_str(), vip_status);
+            throw runtime_error("Failed to remove Vrrp Vip on interface");
+        }
+    }
+  
+    sai_status_t vmac_status = sai_router_intfs_api->remove_router_interface (vrrp_table_[key].rifid);
+    if (vmac_status != SAI_STATUS_SUCCESS)
```

Comment:
If the same address exists, the route entry will not be deleted; however, the router interface below will be deleted, which may cause the route to be bound to a VRRP router interface id that has already been deleted.

Reply:
Same prefix will be present only in VRRP owner case. While adding VIP in owner case, only the RIF is added. Route entry would not be added as the same route will be added because of base interface.

Hence while deleting VIP in owner case, only the RIF will be deleted.